### DISCUSSION TITLE
Why is the Id of Resource in CheckResourcesRequest required?

### DISCUSSION THREAD

**Discussion by bbodensieck** [cerbos/cerbos#1754] [Q&A] [answered]
Hi folks,
we were wondering why the Id of a Resource in the check is required.

Example A: We are creating a new book
```
action := create-book
kind := book:object
id := unknown, because the book does not exist yet
```

Example B: We are checking if a static button should be visible
```
action := view-submit-btn
kind := submit:button
id := submit-button
```

Would you be so kind and tell us the reason why the Id property of the resource is mandatory?

Thank you so much and keep up the great work
Björn

**Comment by charithe** [comment, marked as answer]
Hi. The resource ID is mandatory because Cerbos relies on it for several purposes including scatter-gather operations for handling batch requests and -- most importantly -- recording audit logs. The SDKs also use the ID to provide utility methods for working with batch responses. 

We are aware that there are some situations where the ID is not known. In those cases, our recommendation is to make up an ID that would make sense to someone looking at the audit logs later. For example, `NEW` or `NEW_<session_id>` for new resources.
